### ISSUE TITLE
Server error if no data in date range

### ISSUE THREAD

**Issue by Floppy** [theodi/bothan#136]

**Comment by quadrophobiac** [comment]
So PR [200](https://github.com/theodi/bothan/pull/200) has created a fallback where any date range error results in `No data detected for selected range, please customise date range via the Dates tab below` being presented to the user.

@Floppy I'm trying to get a sense of what the ideal expected behaviour is, what presumptions are safe to make. 
Steps to reproduce
* navigate to https://sport-england-bothan.herokuapp.com/metrics/inactivity, let it default to last 30 days
* navigate to https://sport-england-bothan.herokuapp.com/metrics/inactivity/latest - it should default to Visualisation Type: Chart
* click number radio button in the "`View this as a ..."` tab (you should see most current value for endpoint, you should be able to View JSON, and URL in address should be `https://sport-england-bothan.herokuapp.com/metrics/inactivity/latest?type=number`)
* click chart radio button in the
* you should see an empty chart box (no graceful failure message displayed), you should be able to view JSON) and the URL in address bar should be https://sport-england-bothan.herokuapp.com/metrics/inactivity/2013-01-01%2006:00:00/2017-07-25%2017:43:01?type=chart

Steps: Part 2

* If you navigate directly to https://sport-england-bothan.herokuapp.com/metrics/inactivity/latest?type=chart you should see an empty chart box (no graceful failure message displayed), you should be able to view JSON for latest value

For the link that you've given there's only 3 data points in the entire endpoint, so viewing the entire range of available data isn't a problem. But that won't be the case for every endpoint. It seems like there is a bug in how the chart visualisation handles the type of data returned by the latest endpoint alias. I think this needs to be fixed first so that there can be a generic fallback in place where 
`if JSON.data for last 30 days = 0 then go to endpoint/latest instead`

**Comment by Floppy** [comment]
yeah, tricky. The error message might be enough, but if we do want to do something automatically in chart view, perhaps we should show the last 10 values or something. No date range will work for all metrics - they could be per-second or per-year.

**Comment by quadrophobiac** [comment]
I like that solution of last N values - maybe 30 or 24 (or some other number that maps to longer intervals of time) should be the number.

**Comment by quadrophobiac** [comment]
@Floppy renaming this issue to be more descriptive of what it should resolve, please edit it back if you think its straying from specification

**Comment by quadrophobiac** [comment]
separate ticket #253 opened to investigate whether there is a bug in play
